staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go

```diff
@@ -142,6 +142,34 @@ func (a *Webhook) shouldCallHook(h *v1beta1.Webhook, attr admission.Attributes)
 			break
 		}
 	}
+	if invocation == nil && h.MatchPolicy != nil && *h.MatchPolicy == v1beta1.Equivalent {
+		attrWithOverride := &attrWithResourceOverride{Attributes: attr}
+		equivalents := o.GetResourceMapper().ResourcesFor(attr.GetResource(), attr.GetSubresource())
+		// honor earlier rules first
+		for _, r := range h.Rules {
+			// see if the rule matches any of the equivalent resources
+			for _, equivalent := range equivalents {
+				if equivalent == attr.GetResource() {
+					// we already checked this
```

Comment:
odd optimization. Do you expect many rules with just one resource?

Reply:
we already tried to match attr.GetResource() and failed, no need to check it again

Reply:
> You skip a rule if there is only one equivalent and the previous rule was not matching. Why? Doesn't this mean that you even skip every rule than first rules?

no, we skip checking a particular item in the equivalent list if it is the same as attr.GetResource() (which we already checked above)